src/strands_tools/use_browser.py

```diff
@@ -0,0 +1,1010 @@
+                "required_params": [],
+                "result_template": "New tab created with ID: {result}",
+            },
+            "switch_tab": {
+                "method": lambda tab_id: self._switch_to_tab(tab_id),
+                "required_args": ["tab_id"],
+                "required_params": [("tab_id", str)],
+                "result_template": "Switched to tab: {tab_id}",
+            },
+            "close_tab": {
+                "method": lambda args: self._close_tab_by_id(args.get("tab_id", self._active_tab_id)),
+                "required_args": ["args"],
+                "required_params": [],
+                "result_template": "Tab closed successfully",
+            },
+            "list_tabs": {
+                "method": lambda: self._list_tabs(),
+                "required_args": [],
+                "required_params": [],
+                "post_process": lambda result: json.dumps(result, indent=2),
+                "result_template": "Tabs: {result}",
+            },
+            "get_cookies": {
+                "method": lambda: self._context.cookies(),
+                "required_args": [],
+                "required_params": [],
+                "post_process": lambda result: json.dumps(result, indent=2),
+                "result_template": "Cookies: {result}",
+            },
+            "set_cookies": {
+                "method": lambda args: self._context.add_cookies(args.get("cookies", [])),
+                "required_args": ["args"],
+                "required_params": [("cookies", list)],
+                "result_template": "Cookies set successfully",
+            },
+            "network_intercept": {
+                "method": lambda page, args: page.route(args.get("pattern", "*"), lambda route: route.continue_()),
+                "required_args": ["page", "args"],
+                "required_params": [],
+                "result_template": "Network interception set for {pattern}",
+            },
+            "execute_cdp": {
+                "method": lambda args: self._cdp_client.send(args["method"], args.get("params", {})),
+                "required_args": ["args"],
+                "required_params": [("method", str)],
+                "post_process": lambda result: json.dumps(result, indent=2),
+                "result_template": "CDP {method} result: {result}",
+            },
+            "close": {
+                "method": lambda: self.cleanup(),
+                "required_args": [],
+                "required_params": [],
+                "result_template": "Browser closed",
+            },
+        }
+
+    async def ensure_browser(self, launch_options=None, context_options=None):
+        """Initialize browser if not already running."""
+        logger.debug("Ensuring browser is running...")
+
+        # Ensure required directories exist
+        user_data_dir = os.getenv("BROWSER_USER_DATA_DIR", os.path.join(os.path.expanduser("~"), ".browser_automation"))
```

Comment:
Related to my earlier question - what's our convention with naming?  Should these be prefixed with `STRANDS_`?

Are these parameters only required because we don't provide a way to configure tools otherwise?  What if you implemented it such as:

```
class BrowserTool:
    def __init__(... params...):


    @tool
    def use_browser(...)
           ... 
```


Usage would then be:

```
use_browser = BrowserTool(... params ...).use_browser
```

which would also allow you to isolate state.

(this may or may not work today; but after strands-agents/sdk-python#258 it should work).  We may also want to ping the team to get thoughts on this class based approach for this case